### PR TITLE
Allow emulation version to be set in integration test

### LINKED ISSUE #127639: Emulation Version cannot be set in integration test
Emulation version needs to be configurable both in unit tests and integration tests. Currently, integration tests always emulate the latest version with no other option to change it.

Ref: https://github.com/kubernetes/kubernetes/pull/127302#discussion_r1776095308

/kind bug
/sig api-machinery
/triage accepted
/cc @jpbetz @aaron-prindle

### PR DESCRIPTION
**Pull request by Jefftree**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Allow emulation version to be set in integration test. Use a VAP test as example for creating a disablement test. 

#### Which issue(s) this PR fixes:

Fixes https://github.com/kubernetes/kubernetes/issues/127639

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by cici37** [comment]
/lgtm
Thank you!

File: cmd/kube-apiserver/app/testing/testserver.go [hunk 1]
```diff
@@ -102,8 +103,11 @@ type TestServerInstanceOptions struct {
 	// an apiserver version skew scenario where all apiservers use the same proxyCA to verify client connections.
 	ProxyCA *ProxyCA
 	// Set the BinaryVersion of server effective version.
-	// Default to 1.31
+	// If empty, binary version will be inferred from the build.
```
Comment (jpbetz) [inline cmd/kube-apiserver/app/testing/testserver.go:106]:
```suggestion
	// If empty, effective version will default to version.DefaultKubeBinaryVersion.
```

(To encourage people that need to care to follow the breadcrumbs to the source-of-truth)

**Comment by Jefftree** [comment]
/hold

Hmm noticed the integration test is failing in Cici's PR: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/127302/pull-kubernetes-integration/1839361843622055936. Going to double check this fix before merging

**Comment by cici37** [comment]
> /hold
> 
> Hmm noticed the integration test is failing in Cici's PR: https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/127302/pull-kubernetes-integration/1839361843622055936. Going to double check this fix before merging

Just going to mention it. Thank you!

**Comment by cici37** [comment]
looks like when `SetFeatureGateDuringTest`, the emulation version recognized is using the default version: https://github.com/kubernetes/kubernetes/blob/ff13ffc6e03418201096249cc447e084fcfff268/staging/src/k8s.io/component-base/featuregate/feature_gate.go#L383

**Comment by Jefftree** [comment]
Yeah I see. So we would need to set the feature gate after the emulation version is set, and currently with the PR it's the other way around.

**Comment by Jefftree** [comment]
From talking to Joe, feature gates are generally set before we call StartTestServer so emulation version must be set before that and imo we shouldn't set the emulation version this late in StartTestServer. 

We should also return an error if a feature gate was overridden and we try to set the emulation version after.

**Comment by Jefftree** [comment]
/hold cancel

This is needed for https://github.com/kubernetes/kubernetes/pull/127302. Will address https://github.com/kubernetes/kubernetes/pull/127643#issuecomment-2377977417 in a follow up

**Pull request merged**